File: frontend/src/M3/services/ticketService.js
import axios from "axios";

const API_URL = "http://localhost:8080/api/tickets";

const getAllTickets = () => axios.get(API_URL);
const getTicketById = (id) => axios.get(`${API_URL}/${id}`);
const createTicket = (ticket) => axios.post(API_URL, ticket);
const updateStatus = (id, status) =>
  axios.put(`${API_URL}/${id}/status?status=${status}`);
const addComment = (id, comment) =>
  axios.post(`${API_URL}/${id}/comments`, comment);
const getComments = (id) =>
  axios.get(`${API_URL}/${id}/comments`);
const uploadAttachment = (id, formData) =>
  axios.post(`${API_URL}/${id}/attachments`, formData, {
    headers: {
      "Content-Type": "multipart/form-data",
    },
  });
const assignTechnician = (id, data) =>
  axios.put(`${API_URL}/${id}/assign`, data);
const rejectTicket = (id, data) =>
  axios.put(`${API_URL}/${id}/reject`, data);
const resolveTicket = (id, data) =>
  axios.put(`${API_URL}/${id}/resolve`, data);

const ticketService = {
  getAllTickets,
  getTicketById,
  createTicket,
  updateStatus,
  addComment,
  getComments,
  uploadAttachment,
  assignTechnician,
  rejectTicket,
  resolveTicket,
};

export default ticketService;

File: backend/backend/src/main/java/backend/Module_3/service/TicketService.java
package backend.Module_3.service;

import backend.Module_3.dto.AssignTechnicianRequest;
import backend.Module_3.dto.CommentRequest;
import backend.Module_3.dto.RejectTicketRequest;
import backend.Module_3.dto.ResolveTicketRequest;
import backend.Module_3.model.Attachment;
import backend.Module_3.model.Comment;
import backend.Module_3.model.Ticket;
import org.springframework.web.multipart.MultipartFile;

import java.util.List;

public interface TicketService {
    Ticket createTicket(Ticket ticket);
    List<Ticket> getAllTickets();
    Ticket getTicketById(Long id);
    Ticket updateStatus(Long id, String status);

    Comment addComment(Long ticketId, CommentRequest request);
    List<Comment> getCommentsByTicketId(Long ticketId);

    Attachment uploadAttachment(Long ticketId, MultipartFile file);

    Ticket assignTechnician(Long ticketId, AssignTechnicianRequest request);
    Ticket rejectTicket(Long ticketId, RejectTicketRequest request);
    Ticket resolveTicket(Long ticketId, ResolveTicketRequest request);
}